File: ArrayAlgos/Kadanes algo/sol2.py
def kadanesAlgorithm(array):
    csum = array[0]
    osum = array[0]
    
    for i in range(1,len(array)):
        num = array[i]
        csum = max(num,csum + num)
        osum = max(osum,csum)
    return osum

File: old/ArrayAlgos/Kadanes algo/sol2.cpp
#include <vector>
#include <algorithm>
using namespace std;

int kadanesAlgorithm(vector<int> array) {
  // Write your code here.
    int csum = array[0];
    int osum = array[0];
    for(int i = 1; i < array.size(); i++){
        int num = array[i];
        csum = max(num, csum + num);
        osum = max(osum, csum);
    }
    return osum;
}
